src/pool-bin/libraries/BinPool.sol

```diff
@@ -464,12 +468,26 @@ library BinPool {
     function _subShare(State storage self, address owner, uint24 binId, bytes32 salt, uint256 shares) internal {
         self.positions.get(owner, binId, salt).subShare(shares);
         self.shareOfBin[binId] -= shares;
+
+        /// @dev Ensure bin total share is either 0 or greater than minimum share
+        uint256 balanceShare = self.shareOfBin[binId];
+        if (balanceShare > 0 && balanceShare < MINIMUM_SHARE) {
+            revert BinPool__BelowMinimumShare(balanceShare);
+        }
```

Comment:
only concern is that user maybe will always get revert when try to withdraw all , but left shares of bin is below MINIMUM_SHARE.

Reply:
nice, this would be a con of this approach

So example would be:

1. alice added min_share + 1 liquidity = 1e9 + 1 (pool has 1e9 + 1 share)
2. bob added 10 liquidity  (pool has 1e9 + 1 + 10 share)
3. alice remove 11 liquidity (pool left with 1e9 share)
4. bob is stuck and cannot remove his share

Reply:
per discussion with Snoopy, might worth to consider to limit from the perspective of user position instead of the whole bin.